library/psa_crypto.c

```diff
@@ -7458,14 +7566,25 @@ static psa_status_t psa_pake_complete_inputs(
     psa_crypto_driver_pake_inputs_t inputs = operation->data.inputs;
 
     if (inputs.password_len == 0 ||
-        inputs.role == PSA_PAKE_ROLE_NONE) {
+        inputs.user_len == 0 ||
+        inputs.peer_len == 0) {
         return PSA_ERROR_BAD_STATE;
     }
 
-    if (operation->alg == PSA_ALG_JPAKE &&
-        inputs.role != PSA_PAKE_ROLE_CLIENT &&
-        inputs.role != PSA_PAKE_ROLE_SERVER) {
-        return PSA_ERROR_NOT_SUPPORTED;
+    if (operation->alg == PSA_ALG_JPAKE) {
```

Comment:
This logic rather belongs to `psa_crypto_pake.c` to me as if I understand correctly this is to adapt to the interface of ecjpake.c.  That way we would also already add at the right place the calls to `get_user/peer_id` in `mbedtls_psa_pake_setup()`.

Reply:
Yes, this could be handled in `psa_crypto_pake.c`, but as this is only partial fix I wanted to handle everything at core level. This part will be removed in follow-up PR when driver part will be adapted to use `peer/user`.

Reply:
I agree with @mprse that the goal of this PR is only to fix the user-facing API; we'll need a follow-up for drivers where we can fix this. We've got only two days to get this into the release, and the slow CI doesn't exactly help iterating quickly.

Reply:
okay let's keep as it is.